integration_tests/ttl_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 TiKV Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/tikv/client-go/v2/config"
+	"github.com/tikv/client-go/v2/rawkv"
+)
+
+func TestTTL(t *testing.T) {
+	suite.Run(t, new(ttlTestSuite))
+}
+
+type ttlTestSuite struct {
+	suite.Suite
+	client *rawkv.Client
+}
+
+func (s *ttlTestSuite) SetupTest() {
+	t := s.T()
+	if !*withTiKV {
+		t.Skip("skipping TestTTL because with-tikv is not enabled")
+	}
+	addrs := strings.Split(*pdAddrs, ",")
+	client, err := rawkv.NewClient(context.TODO(), addrs, config.DefaultConfig().Security)
+	require.Nil(s.T(), err)
+	s.client = client
+}
+
+func (s *ttlTestSuite) mustPutWithTTL(key, value []byte, ttl uint64) {
+	err := s.client.PutWithTTL(context.TODO(), key, value, ttl)
+	s.Nil(err)
+}
+
+func (s *ttlTestSuite) mustNotExist(key []byte) {
+	v, err := s.client.Get(context.TODO(), key)
+	s.Nil(err)
+	s.Nil(v)
+}
+
+func (s *ttlTestSuite) mustGetKeyTTL(key []byte) *uint64 {
+	ttl, err := s.client.GetKeyTTL(context.TODO(), key)
+	s.Nil(err)
+	return ttl
+}
+
+// TODO: It's an unstable test, we may mock this feature in unistore.
```

Comment:
Is it unstable now? What about increasing the sleep time.

Reply:
No need. There are very few chances that this test might fail because we can't count on the time of TiKV flow as the same as the clients. The TODO is to TODO for `unistore`.

Reply:
If so, I'd like to remove `It's an unstable test`.